tsdb/db_test.go

```diff
@@ -2434,3 +2436,109 @@ func TestDBReadOnly_FlushWAL(t *testing.T) {
 	testutil.Ok(t, seriesSet.Err())
 	testutil.Equals(t, 1000.0, sum)
 }
+
+// TestChunkWriter ensures that chunk segment are cut at
+// the chunk that is outside the set segment size.
+func TestChunkWriter(t *testing.T) {
+	ch := tsdbutil.ChunkFromSamples([]tsdbutil.Sample{sample{1, 1}})
+
+	tests := []struct {
+		chks             []chunks.Meta
+		segmentSize      int
+		expSegmentsCount int
+	}{
+		// Last chunk ends at the segment boundary so
+		// all chunks should fit in a single segment.
+		{
+			chks: []chunks.Meta{
+				ch,
+				ch,
+				ch,
+			},
+			segmentSize:      chunks.SegmentHeaderSize + 3*len(ch.Chunk.Bytes()),
```

Comment:
`segmentSize` should  include `data length`,  `encoding` and  `crc32`.

Reply:
yep , I also just noticed that and refactored.